src/libs/gltf/gltf_view/lv_gltf.h

```diff
@@ -340,6 +360,70 @@ void lv_gltf_set_antialiasing_mode(lv_obj_t * obj, lv_gltf_aa_mode_t value);
+ * @return camera facing plane
+ */
+lv_3dplane_t lv_gltf_get_current_view_plane(lv_obj_t * obj, float distance);
+
+/**
+ * Get a plane that faces upward, centered at a given height
+ * @param obj pointer to a GLTF viewer object
+ * @param elevation elevation of the ground plane, in world units. this is usually zero
+ * @return ground plane
+ */
+lv_3dplane_t lv_gltf_get_ground_plane(float elevation);
+
+/**
+ * Calculates a ray originating from the camera and passing through the specified mouse position on the screen. It can be used for picking or collision detection with the ground.
+ * @param obj pointer to a GLTF viewer object
+ * @param norm_mouseX normalized mouse x co-ordinate, valid range 0-1
+ * @param norm_mouseY normalized mouse y co-ordinate, valid range 0-1
+ * @return mouse point ray
+ */
+lv_3dray_t lv_gltf_create_ray_from_screen_point(lv_obj_t * obj, float norm_mouseX, float norm_mouseY);
+
+/**
+ * Get the 3d point a mouse co-ordinate would intersect with on a ground plane if projected into the screen
+ * @param obj pointer to a GLTF viewer object
+ * @param screen_x screen x co-ordinate, in pixels
+ * @param screen_y screen x co-ordinate, in pixels
+ * @param value base elevation of the ground plane, this is usually zero
+ * @param obj output lv_3dpoint_t holder, values are only valid if true is the return value
+ * @return true if intersection, false if no intersection
+ */
+bool lv_gltf_raycast_ground_position(lv_obj_t * obj, int32_t screen_x, int32_t screen_y, float base_elevation,
+                                     lv_3dpoint_t * outPos);
+
+/**
+ * Get the 3d point a screen co-ordinate would intersect with on a camera-facing plane at a given distance if projected into the screen
+ * @param obj pointer to a GLTF viewer object
+ * @param screen_x screen x co-ordinate, in pixels
+ * @param screen_y screen x co-ordinate, in pixels
+ * @param offset_distance distance from the view position to set the camera-facing plane at, intersection points will be on this plane
+ * @param outPos output lv_3dpoint_t holder, values are only valid if true is the return value
+ * @return true if intersection, false if no intersection
+ */
+bool lv_gltf_raycast_camera_plane(lv_obj_t * obj, int32_t screen_x, int32_t screen_y, float offset_distance,
+                                  lv_3dpoint_t * outPos);
+
+/**
+ * Get the screen position of a 3d point
+ * @param obj pointer to a GLTF viewer object
+ * @param world_pos world position to convert
+ * @param screen_x the resulting point x coordinate, in pixels. only valid if return value is true
+ * @param screen_y the resulting point y coordinate, in pixels. only valid if return value is true
+ * @return true if conversion valid, false if no valid conversion
+ */
+bool lv_gltf_world_to_screen(lv_obj_t * obj, const lv_3dpoint_t world_pos, int32_t * screen_x, int32_t * screen_y);
```

Comment:
let's have a single output parameter, `lv_point_t`

Also for struct params, pass them as a const reference, not as a value so:

```suggestion
bool lv_gltf_world_to_screen(lv_obj_t * obj, const lv_3dpoint_t * world_pos, lv_point_t * out_point);
```

Reply:
that does not seem realistic, if an invalid co-ordinate is supplied, it will fail

Reply:
?

Reply:
like if the camera is at 0,0,10 looking at 0,0,0 and you request the screen coordinate of 0,0,20 (behind the camera), that will fail.

Reply:
Yes, how does it relate to what I said?

Reply:
oh i see i thought you meant instead of the bool result.  so instead of two ints, use what instead?  lv_point_t?

Reply:
Oh, I see now. Yes, that's what I meant - to use `lv_point_t` instead

Also return type should be `lv_result_t` as I mentioned previously

Reply:
changed in [f33a736](https://github.com/lvgl/lvgl/pull/9143/commits/f33a7362d1aedf02e55c9ff6c75462482ac2abef)